### DISCUSSION TITLE
Audio output giving a distorted alien like sound after training

### DISCUSSION THREAD

**Discussion by VOAdewumi** [coqui-ai/TTS#4189] [General]
Audio output giving a distorted alien like sound after training. I used the glowTTS

This is my test code:
import os
import torch
import gradio as gr
import soundfile as sf
import json
import numpy as np
from TTSMain.TTS.api import TTS
from TTSMain.TTS.vocoder.models.hifigan_generator import HifiganGenerator

device = "cuda" if torch.cuda.is_available() else "cpu"

# Paths to model and vocoder
model_path = os.path.join(os.getcwd(), "tts_train_dir/run-April-11-2025_06+19AM-0000000/best_model_6699.pth")
config_path = os.path.join(os.getcwd(), "tts_train_dir/run-April-11-2025_06+19AM-0000000/config.json")

# Path to HiFi-GAN vocoder
vocoder_model_path = os.path.join(os.getcwd(), "TTSMain/hifi-gan/hifigan_gen_checkpoint_6500.pt")
vocoder_config_path = os.path.join(os.getcwd(), "TTSMain/hifi-gan/config_v2.json")

# Load TTS model
# Load HiFi-GAN vocoder
# Load HiFi-GAN vocoder
print("Loading HiFi-GAN vocoder...")
with open(vocoder_config_path, 'r', encoding='utf-8') as f:
    vocoder_config = json.load(f)

vocoder = HifiganGenerator(
    in_channels=vocoder_config['num_mels'],
    out_channels=1,
    resblock_type=vocoder_config['resblock'],
    resblock_dilation_sizes=vocoder_config['resblock_dilation_sizes'],
    resblock_kernel_sizes=vocoder_config['resblock_kernel_sizes'],
    upsample_kernel_sizes=vocoder_config['upsample_kernel_sizes'],
    upsample_initial_channel=vocoder_config['upsample_initial_channel'],
    upsample_factors=vocoder_config['upsample_rates'],
).to(device)

# Load weights
state_dict = torch.load(vocoder_model_path, map_location=device)
print("Available checkpoint keys:", state_dict.keys())

# Check if EMA weights exist and are not None
if 'gen_ema' in state_dict and state_dict['gen_ema'] is not None:
    print("Loading EMA weights...")
    vocoder.load_state_dict(state_dict['gen_ema'], strict=False)
elif 'generator' in state_dict:
    print("Loading generator weights...")
    # First try strict loading
    try:
        vocoder.load_state_dict(state_dict['generator'])
        print("Loaded with strict=True")
    except RuntimeError as e:
        print(f"Strict loading failed: {e}")
        # Fallback to partial loading
        model_dict = vocoder.state_dict()
        # Filter only matching keys and shapes
        pretrained_dict = {
            k: v for k, v in state_dict['generator'].items() 
            if k in model_dict and v.shape == model_dict[k].shape
        }
        print(f"Matching keys: {len(pretrained_dict)}/{len(model_dict)}")
        model_dict.update(pretrained_dict)
        vocoder.load_state_dict(model_dict)
else:
    raise ValueError("No valid weights found in checkpoint")

vocoder.remove_weight_norm()
vocoder.eval()
print("Vocoder loaded successfully!")

# Output directory
output_dir = os.path.join(os.getcwd(), "audios")
os.makedirs(output_dir, exist_ok=True)

def generate_audio(text="Ìjìyà ò kàn ẹranko tí kò mọ ibi tí a i ń pa ẹ̀sùn"):
    print(f"Generating audio for text: {text}")
    
    try:
        # Generate Mel spectrogram
        print("Generating Mel spectrogram...")
        mel_output = tts.compute_mel(text)
        print(f"Mel shape: {mel_output.shape}")
        
        # Add batch dimension and transpose if needed
        if len(mel_output.shape) == 2:  # (T, n_mels)
            mel_output = mel_output.unsqueeze(0).transpose(1, 2)  # (1, n_mels, T)
        
        # Convert Mel to waveform
        print("Generating waveform...")
        with torch.no_grad():
            mel_output = mel_output.to(device)
            wav = vocoder.inference(mel_output)
        
        # Convert to numpy and normalize
        wav = wav.squeeze().cpu().numpy()
        wav = wav / max(1.0, np.max(np.abs(wav)))
        
        # Save audio
        output_path = os.path.join(output_dir, "output.wav")
        sf.write(output_path, wav, samplerate=22050)
        print(f"Audio saved to {output_path}")
        
        return output_path
    
    except Exception as e:
        print(f"Error during audio generation: {str(e)}")
        return None

# Gradio Interface
demo = gr.Interface(
    fn=generate_audio, 
    inputs=gr.Text(label="Yoruba Text"), 
    outputs=gr.Audio(label="Generated Audio"),
    title="Yoruba TTS Demo"
)

if __name__ == "__main__":
    demo.launch()

**Comment by eginhard** [comment]
It's probably not trained long enough. Try training for 60-600k steps.
